data_transfer/services/inventory.py

```diff
@@ -18,14 +20,27 @@ def device_history(device_id: str) -> Any:
     return response.json()["data"]
 
 
-def patient_id_by_device_id(device_id: str) -> Optional[str]:
-    values = [i for i in device_history(device_id).values()]
+def record_by_device_id(
+    device_id: str, start_wear: datetime, end_wear: datetime
+) -> Optional[Any]:
+    device_wears = [i for i in device_history(device_id).values()]
 
-    if len(values) == 1:
-        return str(values[0]["patient_id"])
-    else:
-        # this means that there are multiple devices in the history, e.g.
-        # that multiple users may have used the same device. Therefore,
-        # we also need to pass in the wear time range and filter by it.
-        # TODO: check which device is within wear time range
-        return None
+    start_wear = utils.normalise_day(start_wear)
+    end_wear = utils.normalise_day(end_wear)
```

Comment:
This is quite important, because are the days inclusive, or exclusive? `Normalise_day()` returns the day at 00:00, which means the end_wear is exclusive. 

I need to watch out for this too, as I was using `datetime.replace(hour=23, minute=59)` for the byteflies implementation. Could use a utils method for this!

For readability, I'd suggest to have two methods, `normalise_day_start` and `normalise_day_end`.

Reply:
I now see your use is different from mine for the ByteFlies implementation. So you can ignore this suggestion.